pymtl3/passes/VcdGenerationPass.py

```diff
@@ -81,6 +79,57 @@ def _gen_vcd_symbol():
         yield code
         n += 1
 
+    # The actual dump_vcd function to be appended to _sched.
+    # The framework will call this function without arguments, and therefore
+    # I chose to monkey patch the data structure objects to the global
+    # namespace of `dump_vcd`. This might be hacky but probably better than
+    # compiling a million lines of code to a Python function (you will almost
+    # certainly get a segfault from that).
+    # TODO: make this less hacky? type check?
+
+    def dump_vcd():
```

Comment:
You already have access to top here. I guess a less hacky way is to write a function that return a function without argument.
```
def vcd_gen(top,vcdfile):
  def dump_vcd():
    # use top/vcdfile
```

Reply:
Yeah good point. I think that's actually the pythonic way how people add some variables to the closure of a function...